### PR TITLE
🔧 add `fee_token_address` in genesis config

### LINKED ISSUE #183: Add fee_token_address in genesis config 🔧

### PR DESCRIPTION
**Pull request by abdelhamidbakhta**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: fixes #183 

# What is the new behavior?

- `fee_token_address` is initialized through genesis config and stored in the pallet

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/primitives/starknet/src/block/serialize.rs [hunk 1]
```diff
@@ -25,15 +26,15 @@ impl SerializeBlockContext for BlockContext {
...
+    fn serialize(block_header: Header, fee_token_address: ContractAddressWrapper) -> BlockContext {
         BlockContext {
             chain_id: ChainId("SN_GOERLI".to_string()),
             block_number: BlockNumber(block_header.block_number.as_u64()),
             block_timestamp: BlockTimestamp(block_header.block_timestamp),
             sequencer_address: ContractAddress::try_from(StarkFelt::new(block_header.sequencer_address).unwrap())
                 .unwrap(),
             cairo_resource_fee_weights: HashMap::default(),
-            fee_token_address: ContractAddress::default(),
+            fee_token_address: ContractAddress::try_from(StarkFelt::new(fee_token_address).unwrap()).unwrap(),
```
Comment (EvolveArt) [inline crates/primitives/starknet/src/block/serialize.rs:37]: could we remove unwraps here ?

File: crates/pallets/starknet/src/lib.rs [hunk 2]
```diff
@@ -201,22 +201,50 @@ pub mod pallet {
     pub(super) type StorageView<T: Config> =
         StorageMap<_, Twox64Concat, ContractStorageKeyWrapper, StarkFeltWrapper, ValueQuery>;
 
+    /// The last processed Ethereum block number for L1 messages consumption.
+    /// This is used to avoid re-processing the same Ethereum block multiple times.
+    /// This is used by the offchain worker.
+    /// # TODO
+    /// * Find a more relevant name for this.
```
Comment (EvolveArt) [inline crates/pallets/starknet/src/lib.rs:208]: LastConsumedL1Block ?

**Review by EvolveArt — CHANGES_REQUESTED** [review]

File: crates/pallets/starknet/src/lib.rs (hunk 2 above)
Comment (abdelhamidbakhta) [reply crates/pallets/starknet/src/lib.rs:208]: yes something like this. but will do it in a follow up PR, I just left the todo not to forget it.

File: crates/primitives/starknet/src/block/serialize.rs (hunk 1 above)
Comment (abdelhamidbakhta) [reply crates/primitives/starknet/src/block/serialize.rs]: done

**Review by EvolveArt — APPROVED** [review]
lgtm

**Pull request merged**